src/Discord.Net.Rest/DiscordRestApiClient.cs

```diff
@@ -803,7 +803,7 @@ internal virtual void Dispose(bool disposing)
             options = RequestOptions.CreateOrClone(options);
 
             var ids = new BucketIds(guildId: guildId);
-            await SendAsync("PUT", () => $"guilds/{guildId}/bans/{userId}?delete-message-days={args.DeleteMessageDays}", ids, options: options).ConfigureAwait(false);
+            await SendAsync("PUT", () => $"guilds/{guildId}/bans/{userId}?delete-message-days={args.DeleteMessageDays}&reason={args.Reason}", ids, options: options).ConfigureAwait(false);
```

Comment:
If we're already doing this at the header level, is there any point in adding the reason to the query string?